### PR TITLE
check the validity of exec agent config

### PR DESCRIPTION
**Pull request by sharanyad**
### Summary
do the same hash comparison check as exec agent log config for the agent config. 

### Testing

New tests cover the changes: yes

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/engine/execcmd/manager_init_task_linux.go [hunk 1]
```diff
@@ -284,8 +284,7 @@ func getAgentConfigFileName(sessionLimit int) (string, error) {
 	configFileName := fmt.Sprintf(execAgentConfigFileNameTemplate, hash)
 	// check if config file exists already
 	configFilePath := filepath.Join(ECSAgentExecConfigDir, configFileName)
-	if fileExists(configFilePath) {
-		// TODO: verify the hash of the existing file contents
+	if fileExists(configFilePath) && validConfigExists(configFilePath, hash) {
```
Comment (angelcar) [inline agent/engine/execcmd/manager_init_task_linux.go:287]: maybe this is for a new PR, but can you check what happens when a directory with the same name as the config file already exists on disk?

My guess is that `createNewExecAgentConfigFile` will fail. I think maybe we need to make sure we remove a directory (or an existing file which is tampered with) before writing the new file

**Review by angelcar — APPROVED** [review]

**Review by ubhattacharjya — APPROVED** [review]

File: agent/engine/execcmd/manager_init_task_linux.go (hunk 1 above)
Comment (sharanyad) [reply agent/engine/execcmd/manager_init_task_linux.go:287]: today we return an error for the following:
- if the file is a directory
- if the file is tampered 

Are you suggesting we do not return an error in these cases?

Comment (sharanyad) [reply agent/engine/execcmd/manager_init_task_linux.go:287]: will change to delete the directory

**Comment by sharanyad** [comment]
`ecs/inf/integ_test ` shows as pending, but have verified from logs that all tests have passed.

**Pull request merged**